src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -197,7 +197,7 @@ private bool ValidateBlobs(Block block, IReleaseSpec spec, out string? error)
     {
         if (spec.IsEip4844Enabled ^ block.ExcessDataGas is not null)
```

Comment:
shouldn't it be `spec.IsEip4844Enabled ^ block.ExcessDataGas is null`?

Reply:
The current one looks correct:
display error if 
IsEip4844Enabled and not (block.ExcessDataGas is not null)
or
not IsEip4844Enabled and block.ExcessDataGas is not null

I agree ^ is confusing sometimes
Replaced with more descriptive variant

Reply:
true^true or false^false -> false
true^false or false^true -> true
We want to return error (so have true^false or false^true) if `IsEip4844Enabled==true && ExcessDataGas is null` or if `IsEip4844Enabled==false && ExcessDataGas is not null`. So it's fine